cpp/src/io/parquet/experimental/hybrid_scan_helpers.cpp

```diff
@@ -554,4 +554,92 @@ aggregate_reader_metadata::filter_row_groups_with_bloom_filters(
   return bloom_filtered_row_groups.value_or(all_row_group_indices(row_group_indices));
 }
 
+/**
+ * @brief Converts column named expression to column index reference expression
+ */
+named_to_reference_converter::named_to_reference_converter(
+  std::optional<std::reference_wrapper<ast::expression const>> expr,
+  table_metadata const& metadata,
+  std::vector<SchemaElement> const& schema_tree)
+{
+  if (!expr.has_value()) { return; }
+
+  // Map column names to their indices
+  std::transform(metadata.schema_info.cbegin(),
+                 metadata.schema_info.cend(),
+                 thrust::counting_iterator<size_t>(0),
+                 std::inserter(_column_name_to_index, _column_name_to_index.end()),
+                 [](auto const& sch, auto index) { return std::make_pair(sch.name, index); });
+
+  // Map column indices to their names
+  auto const& root = schema_tree.front();
+  std::for_each(thrust::counting_iterator<size_t>(0),
```

Comment:
here as well, this looks equivalent with L615

Reply:
Done!